### PR TITLE
Envelope tutorial

### PR DESCRIPTION
**Pull request by benptc**
Can someone quickly skim thru the tutorial and see if it makes sense? Still need to do a bit of formatting of the markdown before I merge, but feedback or approval on the content is useful.

### CODE REVIEW

File: tutorials/envelope-tool-tutorial-part-1.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+### **The simplest envelope**
+
+A simple envelope is included in the default set of tools. This is the simplest possible envelope. It has an icon that you can tap on to open it up and put other tools inside it. Then you can minimize it to hide the contained tools. More interesting demos are possible using the same APIs, but this is a good starting point.
+
+![all-tools-envelope-opening-and-closing](./images/envelope-tool/all-tools-envelope.gif)
+
+### The structure of an envelope
+
+A envelope is really a "system" composed of two different types of tools:
```
Comment (afustePTC) [inline tutorials/envelope-tool-tutorial-part-1.md:34]: An envelope

File: tutorials/envelope-tool-tutorial-part-1.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+An envelope **container** is a tool like the blue square in the above example. It has two states: **open** and **closed**. When it is closed, it just shows up as a small widget floating in 3D space (like most tools). But tapping on it opens it up, which hides that icon and has the option to show some additional 2D graphics on the screen (like the X button in the corner).
+
+When an envelope container is open, and a compatible tool is dropped into the scene, the newly added tool will get saved as one of that container's **contents**. All the envelope contents will be visible until the envelope is closed.
+
+An envelope container gets to choose which tools can go inside it. In the above example, the blue envelope tool has a list of compatible contents, including the on and off buttons. But you can create custom types of envelopes that can only contain specific tools that know how to work together.
+
+### Example Code 1: simple-envelope-container
+
+Explaining everything about envelopes on a conceptual level without looking at any code examples can be confusing. Let's build a some envelope tools in a new add-on to explore how they work.
```
Comment (afustePTC) [inline tutorials/envelope-tool-tutorial-part-1.md:46]: typo: let's build some

File: tutorials/envelope-tool-tutorial-part-1.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+Start out by creating a new add-on called `envelope-tutorial-addon` and creating a `tools` directory in it for the tools we're about to build. If you have any trouble setting it up, refer back to the [Creating a Simple Tool](../develop/spatial-tools/tutorial) tutorial for more detailed instructions.
+
+Create a new tool there named `simple-envelope-container`. The resulting file path that should should end up with for your tool's HTML is `vuforia-spatial-edge-server/addons/envelope-tutorial-addon/tools/simple-envelope-container/index.html`
```
Comment (afustePTC) [inline tutorials/envelope-tool-tutorial-part-1.md:50]: typo: that should end up

File: tutorials/envelope-tool-tutorial-part-1.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+6. **Add a touch event to switch between open and closed states**
+
+```javascript
+rootElementWhenClosed.addEventListener('pointerup', function() {
+    envelope.open();
+});
+```
+
+Lastly, we add a touch event listener to the div that's visible when the envelope is closed. When we tap on it, we used one of the envelope APIs (`envelope.open()`) to open the envelope. Doing this hides rootElementWhenClosed, shows rootElementWhenOpen, and makes visible any tools contained by this envelope.
```
Comment (afustePTC) [inline tutorials/envelope-tool-tutorial-part-1.md:205]: typo? 'we use one of the envelope'

**Comment by benptc** [comment]
> Sounds good to me Ben! I was going to ask about linking the tools inside with outside tools but I guess this would come in part 2 of the tutorial

Thank you for catching all those typos Anna! I could add a short section on the bottom that shows how to link with outside tools. But yes a lot more will be explained in Part 2

**Comment by benptc** [comment]
I added a section about linking envelope nodes to the bottom in response to feedback - thanks!

**Pull request merged**